src/lib/storage/chunk.cpp

```diff
@@ -15,26 +15,37 @@
 namespace opossum {
 
 void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) {
-  // Implementation goes here
+  _segments.push_back(segment);
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == column_count(), "Given value count does not match column count");
+
+  auto value_it = values.cbegin();
+  auto value_end = values.cend();
+  auto segment_it = _segments.cbegin();
+  while(value_it != value_end) {
+    (*segment_it)->append(*value_it);
+
+    ++value_it;
+    ++segment_it;
+  }
 }
 
 std::shared_ptr<BaseSegment> Chunk::get_segment(ColumnID column_id) const {
-  // Implementation goes here
-  return nullptr;
+  return _segments.at(column_id);
 }
 
 uint16_t Chunk::column_count() const {
-  // Implementation goes here
-  return 0;
+  return _segments.size();
 }
 
 uint32_t Chunk::size() const {
-  // Implementation goes here
-  return 0;
+  if (!_segments.size()) {
```

Comment:
I would prefer `_segments.size() == 0` here as it is more semantic.
Opinions @XPerianer @leogeier